kobo/settings/base.py

```diff
@@ -349,6 +349,38 @@
         'having the system empty it automatically.',
         'positive_int_minus_one',
     ),
+    # Enable the use of validators with custom rules
+    'PASSWORD_VALIDATION_CUSTOM_RULES_ENABLED': (
+        True,
+        'Enable custom password validation rules. '
+        'The subsequent rules will not work unless this field is Enabled',
+    ),
+    # Built-in password validator
+    'MINIMUM_LENGTH_VALIDATION': (
+        10,
+        'Minimum length for all passwords',
+    ),
+    # Build-in password validator
+    'USER_ATTRIBUTE_SIMILARITY_VALIDATOR': (
+        True,
+        "Custom user attribute similarity validation is enabled",
+    ),
+    # Built-in password validator
+    'COMMON_PASSWORD_VALIDATION': (
+        '',
+        'Add to list of common passwords for password validation',
+    ),
+    # Custom password Validator
+    'CUSTOM_CHARACTER_CLASSES_ENABLED': (
+        '^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]',
```

Comment:
Are we able to fulfill the requirements with single line regex? My regex knowledge is not strong, but I think it wouldn't be possible to have multiple classes **and** "3 out of 4" feature. 

Maybe we could have a list of classes instead? Something like (needs better names probably):

```
character_classes_validation_minumum_true: 3,
character_classes_validation: [
  '<first regex>',
  '<second regex>',
  '<third regex>',
  '<fourth regex>'
]
```

Reply:
I'm thinking that it would be ideal if all these singular regexes could be described in admin UI. So if superuser adds `<some regex>` they could describe it, e.g.

```
…classes: [
  ['Uppercase letters', '<regex>']
]
```

This way it would be easier to remember what each regexes is supposed to do. Plus it would give us opportunity to give some default values as examples - either as actual default or in documentation

Reply:
Sure, I like this idea